wsclient/src/main/java/org/hpccsystems/ws/client/platform/DropZone.java

```diff
@@ -169,36 +180,39 @@ protected void fullRefresh()
     {
         try
         {
-            HPCCFileSprayClient fileSprayClient = platform.getFileSprayClient();
-            PhysicalFileStruct[] response = fileSprayClient.listFiles(getIP(), getDirectory(), getOS());
-            update(response);
+            update(platform.getWsTopologyClient().queryDropzoneMachines(dzInfo.getName()));
         }
         catch (Exception e)
         {
             e.printStackTrace();
         }
     }
 
-    public void update(TpDropZone dz)
+    private void update(TpMachine[] queryDropzoneMachines)
     {
-        if (info.getName().equals(dz.getName()))
-        {
-            info = dz;
-            setChanged();
-        }
+         if (queryDropzoneMachines != null)
+         {
+             machines.clear();
+             for (TpMachine machine : queryDropzoneMachines)
+             {
+                 PhysicalMachine physicalMachine = getMachine(machine.getName());
+                 if (physicalMachine != null)
+                 {
+                     physicalMachine.update(machine);
+                     machines.add(physicalMachine); // Will mark changed if needed ---
+                 }
+             }
+         }
     }
 
-    synchronized boolean update(PhysicalFileStruct[] rawFileStructs)
+    public void update(TpDropZone dz)
     {
-        if (rawFileStructs != null)
+        if (dzInfo.getName().equals(dz.getName()))
```

Comment:
Should this be `if (dzInfo.getName().equals(dz.getName()) == false)` ?

Reply:
the update functions are supposed to update the equivalent object.